---
author: The OBF
date: "2018-12-11T19:11:44+00:00"
guid: https://www.open-bio.org/wp/?page_id=18
title: Projects
url: /projects/

---

The OBF has [a process for different projects in the open bioinformatics real to become members or affiliated members](/adding-projects/). Here you can find the existing member projects, affiliated projects and other related open bioinformatics projects.

# Member Projects


* **[BioJava](http://biojava.org/)** is an open-source project dedicated to providing a Java framework for processing biological data. It provides analytical and statistical routines, parsers for common file formats and allows the manipulation of biological sequences and 3D structures. The main goal of the project is to facilitate rapid application development for bioinformatics.
* The **[Bioperl Project](http://bioperl.org/)** is an international association of users & developers of open source [Perl](https://en.wikipedia.org/wiki/Perl) tools for bioinformatics, genomics and life science BioPerl is a collection of Perl modules that facilitate the development of Perl scripts for bioinformatics applications. It has played an integral role in the [Human Genome Project](https://en.wikipedia.org/wiki/Human_Genome_Project).
* [**BioPython**](https://biopython.org/) is a set of freely available tools for biological computation written in Python by an international team of developers. It is a distributed collaborative effort to develop Python libraries and applications which address the needs of current and future work in bioinformatics. The source code is made available under the [Biopython License](https://github.com/biopython/biopython/blob/master/LICENSE.rst), which is extremely liberal and compatible with almost every license in the world.
* [**BioRuby**](http://www.bioruby.org/) is a collection of open-source Ruby code, comprising classes for computational molecular biology and bioinformatics. It contains classes for DNA and protein sequence analysis, sequence alignment, biological database parsing, structural biology and other bioinformatics tasks.
* [**BioSQL**](https://biosql.org/wiki/Main_Page) is a generic unifying schema for storing sequences from different sources, for instance Genbank or Swissprot. BioSQL is meant to be a common data storage layer supported by all the different Bio* projects, Bioperl, Biojava, Biopython, and Bioruby. Entries stored through an application written in, say, Bioperl could be retrieved by another written in Biojava.

# Affiliated Projects

* [**GBrowse**](https://gmod.org/) (Generic Genome Browser), a part of GMOD is a combination of database and interactive web pages for manipulating and displaying annotations on genomes.
* [**Chado**](https://gmod.org/wiki/Chado) is a relational database schema that underlies many GMOD installations.

# Other Related Projects

* [BioJS](http://biojs.net/): An open-source library of JavaScript components to visualise biological data.
* [BioHaskell](http://biohaskell.org/): Bioinformatic libraries in Haskell.
* [Biocaml](http://biocaml.org/): Bioinformatics library in the functional programming language OCaml.
* [BioPHP](http://biophp.org/): Collection of open-source PHP code, with classes for DNA and protein sequence analysis, alignment, database parsing, and other bioinformatics tools.
* [BioSmallTalk](https://biosmalltalk.github.io/): Bioinformatics in a pure objects environment.
* [CGL](https://www.yandell-lab.org/software/cgl.html): Comparative Genomics Library.
* [BioMake & Skam](https://skam.sourceforge.net/): Task management tool, similar to makefiles.

# Retired Projects


* BioSOAP: Aimed to support biological analysis pipelines remotely via SOAP from BioPerl, BioJava and Ensembl initially.
* BioCORBA: Aimed to support biological analysis pipelines remotely via CORBA and a choice of programming languages.
* [MOBY](http://biomoby.open-bio.org/): A system for interoperability between biological data hosts and analytical services.
* [OBDA](http://obda.open-bio.org/): Standardization of access to sequence data resources.
* Wikiomics: Bioinformatics Wiki.
* [Pise](https://pubmed.ncbi.nlm.nih.gov/12511068/): A tool to generate Web interfaces software for submitting jobs to these interfaces.
* OpenBQS: Bibliographic query system.
* BioClipse: Visual platform based on Eclipse.
* [BioWeka](https://sourceforge.net/projects/bioweka/): Adds bioinformatics functionalities such as e.g. alignments to the popular machine learning framework Weka.
* [BlipKit](https://github.com/cmungall/blipkit): Chris Mungallâ€™s Prolog toolkit for Bioinformatics and BioMedical Informatics.
* [Biolib](https://github.com/biolib/obf-biolib): Cross-project/language C bindings for a shared codebase across the different Bio* projects (Pjotr Prins).
* DAS / BioDAS: The Distributed Annotation System defines a communication protocol used to exchange annotations on genomic or protein sequences.
* [EMBOSS](https://emboss.sourceforge.net/): An analysis package developed for the needs of the molecular biology (e.g. EMBnet) user community. 
