---
author: armando_djiyou
category:
  - event-fellowship
  - travel-fellowship
date: "2020-09-24T08:06:10+00:00"
guid: https://www.open-bio.org/?p=5079
tag:
  - africa
  - bioinformatics
  - cameroon
  - open-life-science
  - open-source
title: 'H3ABionet’s Introduction to Bioinformatics Training (IBT) 2020: a report of the first iteration hosted in Cameroon'
url: /2020/09/24/h3abionet-course-2020-armando-blondel/

---
_The [Open Bioinformatics Foundation (OBF) Event Fellowship program](/travel-awards) aims to promote diverse participation at events promoting open source bioinformatics software development and open science practices in the biological research community. Armando Blondel DJIYOU DJEUDA, a researcher from University of Douala, Cameroon, set up and host [H3ABioNet‘s Introduction to Bioinformatics course (IBT)](https://www.h3abionet.org/categories/training/introduction-to-bioinformatics-training-2020), supported by this fellowship granted to him in application round 1 of 2020._

Due to the rich biodiversity of sub-Saharan African countries, bioinformatics and data management are crucial in helping to advance biomedical research. While many African countries are emerging in those specific fields, Cameroon and Central African countries in general, are still lagging behind. Based on this observation, in 2019, we created Bioinformatics and Biostatistics Hub (B2Hub) at the Biotechnology Center of the [University of Yaounde I](https://uy1.uninet.cm/). Our aim through this platform is to promote bioinformatics and biostatistics among graduate students and young researchers from Central Africa.

After successfully organizing a [first Bioinformatics workshop in 2019](https://lapher-biotech.net/2020/02/13/b2hub-bioinformatics-and-biostatistics/), this year I set up a virtual classroom for the [H3ABioNet‘s Introduction to Bioinformatics course (IBT)](https://www.h3abionet.org/categories/training/introduction-to-bioinformatics-training-2020) at the [Laboratory for Public Health Research Biotechnologies](https://lapher-biotech.net/), Biotechnology Center, [University of Yaounde I](https://uy1.uninet.cm/). I decided to host this event because no other host was selected in Cameroon (even in Central Africa) last year and as a result, I and most likely many other interested Central African candidates were unable to attend this course.

The workshop ran from May to August 2020 (see [full curriculum](https://www.h3abionet.org/categories/training/introduction-to-bioinformatics-training-2020)). In this blog post, I have provided a monthly summary of how this workshop was conducted. I have also provided the demographic distribution of our attendees, their learning process, personal and COVID-19 related challenges experienced by them, and the overall impact of this event.

### **Training Format**

The training was carried out using a distance-based learning model where classrooms were set up at different sites (a total of 47 sites with more than 1000 participants this year) and linked to the trainer via Zoom conferencing software during biweekly contact sessions. The trainers used a combination of theoretical and practical sessions for participants to gain practical experience in using various tools and resources. Initially, the course was supposed to run from 10:30 am to 2:30 pm every Tuesday and Thursday for 3 months (12th May 2020 to the 27th August 2020) at the Biotechnology Center, University of Yaounde I, Cameroon. However, due to the current Covid-19 pandemic, it was no longer possible to have contact sessions and the training was then shifted entirely online. In our classroom, we decided to keep in contact with our participants through online meetings every week and we created a WhatsApp group to allow participants to ask questions directly to our teaching assistants and exchange each other.

As a consequence, the course was highly demanding particularly in terms of internet connection and other basic equipment to conduct online meetings (such as webcam, headsets, small hardware needs, or a TV screen for practicals), especially in our context where it is not yet widely accessible. Thanks to the support of the [Open Bioinformatics Foundation (OBF)](/) through the OBF (travel) fellowship, I was able to get those equipment and successfully host our first IBT course.

### **May: Introduction to Bioinformatics Resources**

We started our journey in May with a short introduction about Bioinformatics and its relevance in biomedical research, before moving to module 1: Introduction to databases and resources. In this module, we learned with Dr Shaun Aron how to navigate through biological databases and resources (like [NCBI](https://www.ncbi.nlm.nih.gov/) or [EBI](https://www.ebi.ac.uk/)), perform basic protein and DNA analysis (sequence features, start sites, restrictions sites and so on), and conduct a more efficient search using [PubMed](https://pubmed.ncbi.nlm.nih.gov/).

### **June: Unix Command-line and Sequence Alignment Theory**

After a promising first month, we covered two exciting modules in June. We started with an introductory workshop to the Linux/UNIX command-line, which was delivered by Dr. Amel Ghouila. Using an emulator called [Cygwin](https://www.cygwin.com/) (for Microsoft users) or Ubuntu (for MAC OS users), we got useful commands and tips to manipulate files and directories. Most of the participants were afraid because it was their first time working on Linux and its command-line environment. However, Dr. Ghouila was able to ease the learning process of beginners to command line computing. We then moved to the third module: Sequence alignment theory and application. In this module, after describing the most popular algorithms used for both global and local alignment, Dr. Jonathan Kayondo then taught us how to use [BLAST](https://blast.ncbi.nlm.nih.gov/) and other pairwise sequence alignment tools (such as [EMBOSS water](blank) and [EMBOSS needle](blank)) to answer some biological questions.

### **July: Multiple Sequence Alignment and Genomics**

In July, we deepened our knowledge and skills in sequence alignment, especially using multiple sequence alignment (MSA) tools and software. We learned the main characteristics of the most popular programs for building MSA and how to select the appropriate one to use. Indeed, we saw the differences between Cobalt, ClustalW, MUSCLE, and Tcoffee, and we used mainly online tools during our practical sessions. Besides, we had an initiation to [Jalview](https://www.jalview.org/), an exceptional multiple sequence alignment viewer, editor, and analysis tool. Dr. Ahmed M. Alzohairy showed us how to use Jalview for editing and analyzing MSAs produced with the most popular MSA programs we saw before.

We then continued during the same month with one of the most attractive modules in my opinion: Genomics. We started with a short overview of sequencing technologies and NGS and further proceeded with more complex topics such as genome annotation, comparative genomics, and human variation. During this module, we used the [Ensembl genome browser](https://www.ensembl.org/index.html), an amazing platform that allows access to publicly available reference genomic data such as genes (through tools and a visual interface), variation data, and annotation of your variants through the [Variant Effect Predictor (VEP) tool](https://www.ensembl.org/vep#:~:text=VEP%20determines%20the%20effect%20of,as%20well%20as%20regulatory%20regions.&text=SIFT%20and%20PolyPhen%2D2%20scores,...). We closed this module by applying some basic Linux command-lines we learned in module 2, to visualize our data and retrieve some basic information from our sample sequences.

### **August: Molecular Evolution and Phylogenetics**

Finally, we entered into our last month (August) with the last module: Molecular evolution and phylogenetics. This module, entirely taught by Dr. Anna Zhukova, was full of new skills, tools, and resources. Briefly, we were introduced to the components of a phylogenetic tree, discussed the basic principles of building a tree, different types of methods (Distance method, max parsimony, and Max likelihood), and how to interpret a phylogenetic tree. During practical sessions, we build our trees using both an online tool ( [NGPhylogeny](https://ngphylogeny.fr/tools/)) and a downloadable software tool ( [Seaview](http://doua.prabi.fr/software/seaview)). We also used [iTOL](https://itol.embl.de/upload.cgi) for tree visualization and manipulation tools and [SMS](http://www.atgc-montpellier.fr/sms/) for the selection of the best evolutionary model. We end up training with tree dating and phylogeography. During this session, we used [TempEst](http://tree.bio.ed.ac.uk/software/tempest/), a downloadable software tool to check for temporal signals in the tree, [LSD/LSD2](https://lsdating.pasteur.fr), an online tree dating and rooting tool, and [PastML](https://pastml.pasteur.fr/) for ancestral character reconstruction.

### **Demographic Distribution of Workshop Attendees**

A total of 44 participants were selected in our classroom (41% were female), mainly coming from Cameroon. We also enrolled 3 participants coming from neighboring countries, especially Chad. They were mostly MSc and PhD students (87.5%) but also post-doc or young researchers (10%). At the end of the training, we build a [post-training evaluation form using a Google forms questionnaire](https://docs.google.com/forms/d/1WqH5i9xr83P4KC0HhgEOAyhnvWUe63nB0siwTyLRzms/edit#question=803302123&field=1644277189) to obtain feedback from participants. The majority (82%) of our participants found the information and/or skills gained during this training highly relevant and useful in their daily activities, 86% of them were satisfied with the course design and methodology and will therefore highly recommend this training course to their classmates and/or colleagues. The dark point of our training was the low completion rate.

Indeed, only 15 participants out of 44 have completed the entire training and obtained a certificate. The most evocated reason among those who did not complete the course was the lack of internet connection at home to complete their assignments and they had not enough time to complete the training. However, even those who did not complete the training admitted having improved their bioinformatics skills and will be happy to complete the training during the next edition as long as we maintain contact sessions as initially planned.

### **Positive Feedback**

I would also like to end up by sharing with you some general comments and appreciation by our participants:

> “ **_This training was as I expected. This allowed me to use many tools and computer software to analyze and annotate different types of biological data as well as to retrieve information from databases and more... This gave me the level I needed to start my own analyzes related to my PhD research._**” Brice Fonkou NATCHEMA SOH, PhD candidate, Faculty of Sciences, University of Yaoundé I, Cameroon.

> “ **_The training was great and understandable for a beginner like me. Haven't done that before, it was really challenging but finally relevant._**” Nelly Ornelle ONANA ATEBA, PhD candidate, Faculty of Sciences, University of Yaoundé I, Cameroon.

> “ **_It was a good experience. I loved being part of it. I will do it again if the opportunity comes to me._**” Pacome Valery TCHUENKAM KOM, PhD candidate at the Dschang School of science and technology, University of Dschang, Cameroon.

> “ **_The course was very enlightening and instructive, with great insights that will definitely be helpful in my current and future research avenues._**” Willy Leroi TOGNA PABO, PhD candidate at the University of Buea, Cameroon.

To conclude, I would like to acknowledge the OBF for providing a fellowship to host an IBT classroom here in Cameroon. Thanks to this fellowship, we will now be able to offer an Introduction to Bioinformatics course to interested students and young graduates every year. This is a great step in the process of expanding bioinformatics skills and knowledge in an underrepresented region like Central Africa. Encouraged by the success of this year’s workshop, we are now considering hosting intermediate courses and building an open learning circle in our classroom. Feel free to get in touch if you would like to stay updated on this activity (email: armando\_djiyou@yahoo.fr).
