---
author: jemima.becker
category:
  - community
  - event-fellowship
  - events
  - travel-fellowship
  - uncategorized
cover:
  alt: jemima-becker - 1
  image: /wp-content/uploads/2022/06/jemima-becker-1.jpeg
date: "2022-06-23T22:04:41+00:00"
guid: https://www.open-bio.org/?p=6510
title: My trip to the EMBO neural stem cells workshop
url: /2022/06/23/obf-fellow-jemimabecker2022/

---
_The [Open Bioinformatics Foundation (OBF) Event Fellowship program](/travel-awards) aims to promote diverse participation at events promoting open source bioinformatics software development and open science practices in the biological research community. Jemima Becker, DPhil student at Merton College, University of Oxford, attended the [EMBO workshop: “Neural stem cells: From basic understanding to translational applications"](https://meetings.embo.org/event/21-neural-stem-cells), supported by this fellowship granted to him in the first round of 2022._

This June, I had the opportunity to travel to the EMBO workshop: “Neural stem cells: From basic understanding to translational applications”. There, I presented a poster documenting the first six months of my PhD work on long noncoding RNAs (lncRNAs) in the postnatal ventricular-subventricular (V-SVZ): “Long noncoding RNAs in the Ventricular-Subventricular zone: what have we learnt from single cell transcriptomics?”.

![](https://lh5.googleusercontent.com/z-pNd6B0_0jByEfOAgydy_CLhx33lUvdLoB-x_dKsNto7HKwHjimxJ0IcRb1fwg0DOI0XuiD00Jk2IgyQQD0s09XtR6prjJ7ysguMKluXSSfXWWxJTWJd502XxCzaqXXfqsMbJqo_fxrjtK1rg)

![](/wp-content/uploads/2022/06/jb1.png)

_Images: My supervisor Francis Szele and I after the end of the first poster session (left). Sunset on the first evening at the conference (right)._

This work was a broad meta-analysis of single cell RNAseq papers, in conjunction with ChIP-seq data in order to identify candidate genes for functional studies into the role of V-SVZ lncRNAs. I presented my computational logic and workflow for this study, linking back to how open access data and code can be used to construct a robust prediction of regulatory genes. I made the argument that, particularly in the stem cell field, a lot of redundant data is generated; during the process of this project I found many overlapping studies – by making use of online data to identify candidates for further study we can save time, expense, and resources by narrowing down the pool of molecules to investigate in vitro or in vivo. Through the results of this analysis, I have started mechanistic investigations into two select lncRNAs and the role they may be playing in V-SVZ neurogenesis in vivo.

The conference was an invaluable experience, as it provided me with not only the opportunity to meet and discuss my work with others working in high resolution’ omics, but also allowed me to network with researchers tackling similar problems using divergent approaches.

**Day 1**

The first day of the conference began with an early flight from London to Athens, followed by a coach journey to a beautiful venue on the coast. The first keynote talk of the conference was given by Malin Parmar, discussing the applicability and translational capacity of stem cell studies in treatment of human neurodegenerative disease. A fascinating aspect of this talk was the use of scRNAseq to analyse the survival and development of tissue grafts, linking high resolution molecular description to gross changes in motor performance.

**Day 2**

On my second day in Kyllini, I attended four sessions totalling 15 talks and was able to present my poster in the evening. Of great excitement to me was a talk from Ana Martin-Villalba regarding a single cell triple-omics approach to analyse stem cell trajectories. The first of two poster sessions took place this evening from 8:30pm, and I found enough people interested in discussing my approach that I stayed at the session until midnight! It was incredibly valuable to be able to talk to other people, as many approaches to finding novel regulatory genes in the V-SVZ (or other tissues) take into account a limited amount of data; by comparing different datasets to identify consistent trends I had been able to find a large number of candidate genes, and was able to discuss the strengths and difficulties of my strategy.

**Day 3**

The third day focused on bioengineering of neural stem cells and 3D human models of neural stem cells. One talk that I found particularly interesting was from Marisa Karow, which focused on the utilisation of SNP maps to identify the allelic origin of individual transcripts in sequencing data. I have previously worked on this with mouse-derived tissue, although I found difficulties with my approach when examining low-abundance transcripts such as lncRNAs, so hearing about Professor Karow’s strategy was very useful.

**Day 4**

The fourth and final day of the conference consisted of three sessions, with a focus on translational capacity and patient care. I enjoyed a talk from Noelia Urban Avellaneda examining the spectrum of stem cell states within the V-SVZ. This was particularly relevant to my work, as something I have noticed in the papers whose scRNAseq data I have examined is that the number of and criteria for different V-SVZ stem cell states is apparently defined on a paper-by-paper, rather than universal basis. Dr Urban Avellaneda proposed a fluid model in which stem cells exist within a complex topology, rather than a single trajectory, of developmental potential.

Overall, my first conference has been an unforgettable opportunity and a fantastic introduction to the wider world of neural stem cells. An especially valuable element, for me, was the chance to meet more women working in bioinformatics – the bulk of my interactions within local computational biology as an undergraduate consisted of male-dominated classes, a trend which has persisted through my classes and workshops as a graduate student. The combination of meeting a greater diversity of scientists alongside the chance to share and discuss my ideas was highly rewarding, and I look forward to taking up more opportunities like this in the future.

I would like to express my gratitude to the Open Bioinformatics Foundation for making this experience possible, and thank them for their support.
