---
author: hilmar
category:
  - blogroll
  - community
  - travel-fellowship
date: "2017-08-28T02:21:27+00:00"
guid: https://news.open-bio.org/?p=1743
summary: _This blog post is syndicated from a [post on Vitalina Kirgizova's blog](https://vitalinabiology.com/2017/06/06/bioinformatics-training-course/), originally published June 6, 2017. Vitalina was supported by the ongoing [OBF travel fellowship program](https://github.com/OBF/obf-docs/blob/master/Travel_fellowships.md) to attend bioinformatics training course held at the Instituto Gulbenkian de Ciência, Oeiras, Portugal, in November 2016._ _The OBF's Travel Fellowship program continues to help open source bioinformatics software developers with funding to attend conferences, workshops, or training events. The next call closes 15 December 2017._
tag:
  - community
  - travel-fellowship
title: OBF Travel Fellowship - IGC Bioinformatics Training
url: /2017/08/27/obf-travel-fellowship-vitalina-kirgizova/

---
_This blog post is syndicated from a [post on Vitalina Kirgizova's blog](https://vitalinabiology.com/2017/06/06/bioinformatics-training-course/), originally published June 6, 2017. Vitalina was supported by the ongoing [OBF travel fellowship program](https://github.com/OBF/obf-docs/blob/master/Travel_fellowships.md) to attend bioinformatics training course held at the Instituto Gulbenkian de Ciência, Oeiras, Portugal, in November 2016._ _The OBF's Travel Fellowship program continues to help open source bioinformatics software developers with funding to attend conferences, workshops, or training events. The next call closes 15 December 2017._

Better late than never. On November 2016 I participated at the training course- [Analysis of Adaptive Immune Receptor Repertoires using high throughput sequencing data (NGS)](ftp://gtpb.igc.gulbenkian.pt/bicourses/2016/AAIRR16/index.html). Many thanks for invitation for [Pedro Fernandes](http://gtpb.igc.gulbenkian.pt/bicourses/index.html), who organizes bioinformatic training courses during 20 years! On the basis of the Instituto Gulbenkian de Ciência, Oeiras, Portugal we could go deep into the jungle of algorithms for the analysis of antibody and TCR genes. Our instructor Dr [Simon Frost](http://www.vet.cam.ac.uk/directory/sdf22@cam.ac.uk) from University of Cambridge, U.K., made journey exciting and interesting.

And I specially want to thank [Open Bioinformatics Foundation](https://news.open-bio.org/about/) for supporting my participation!

![img_0184](https://vitalinabiology.files.wordpress.com/2016/12/img_0184.jpg?w=700)

At the IGC bioinformatic class. (Ph Pedro Fernandes)

Antibodies, also called immunoglobulins, are large Y-shaped proteins. They are produced by B -cells which function to identify and help remove foreign targets such as viruses, bacteria without damaging the rest of the body.

The T-cell receptor, or TCR, is a molecule found on the surface of T cells, that is responsible for recognizing fragments of antigen as peptides bound to major histocompatibility complex (MHC) molecules. T-lymphocytes protect our body from cancer invasions and viruse infections.

Each antibody or TCR binds one unique molecule, just as the key fits into the lock. A human can probably make more than 1 000 billion different adaptive immune receptors molecules – it is adaptive immune receptor repertoire.

![04-t-limfocit-i-kljuch](https://vitalinabiology.files.wordpress.com/2016/12/04-t-limfocit-i-kljuch.png?w=700)

Cute T-lymphocyte holds key – T-Cell Receptor (TCR). (Illustration- me)

Immunoglobulins are proteins, and proteins are encoded by genes. Randomized combination inherited gene segments – V(D)J recombination of DNA and somatic hypermutation are the primary mechanisms diversification of the human antibody repertoire. High-throughput sequencing (HTS) enables thorough investigation of the diverse immune receptors that determine the specificity of adaptive immunity responses.

![14947862_1110768205708694_5847757213665972751_n](https://vitalinabiology.files.wordpress.com/2017/06/14947862_1110768205708694_5847757213665972751_n.jpg?w=700)

Sweet immuno- bioinformatic November

I had an experience in repertoire analysis before. I use software developed at the laboratory where I’m currently working [MiLaboratory](https://milaboratory.com). Also I coauthored a paper reporting the software for immune receptor analysis [VDJtools](https://vitalinabiology.com/publications/).

![img_0185-1](https://vitalinabiology.files.wordpress.com/2016/12/img_0185-1.jpg?w=700)

Our international team. Dr Simon Frost (course instructor) at the left. (Ph Pedro Fernandes)

Nevertheless practical course give me an opportunity to develop my knowledge, try different solutions and protocols: IgBLAST, pRESTO, Change-O, etc.,  Also this was the first time when I worked with very useful software tool [Jupyter Notebook](http://jupyter.org). It helps to manage the code and looks very attractive for bioinformatician who grew up from a wet-laboratory (i.e. for me).

I have a great time at Portugal. The community of the researchers whom use the immune receptor repertoire analysis in daily routine is not very large. This is very specific and complicate topic because of peculiar nature of the lymphocyte receptor rearrangement. So bioinformatic course was a perfect opportunity for me to accumulate knowledge directly from highly skilled professionals.

![Snapseed (1)](https://vitalinabiology.files.wordpress.com/2017/06/snapseed-1.jpg?w=700)

After another one hard day at the bioinformatic course. With Simon Frost and Pedro Fernandes, bioinformatics training coordinator, at the left)

[![](http://feeds.wordpress.com/1.0/comments/vitalinabiology.wordpress.com/868/)](http://feeds.wordpress.com/1.0/gocomments/vitalinabiology.wordpress.com/868/)![](https://pixel.wp.com/b.gif?host=vitalinabiology.com&blog=121300637&post=868&subd=vitalinabiology&ref=&feed=1)
